### PR TITLE
Improve release invocation

### PR DESCRIPTION
**Pull request by rgrinberg**
* Run dune-release tag
* Make sure that dune-release uses the in source dune

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -126,9 +126,15 @@ dune: $(BIN)
 	$(BIN) $(RUN_ARGS)
 
 .PHONY: default install uninstall reinstall clean test doc dev-switch
-.PHONY: promote accept-corrections opam-release dune check fmt
+.PHONY: promote accept-corrections release opam-release dune check fmt
 
-opam-release:
+# Use this target to make sure that we always run the in source dune when making
+# the release
+release: dev
```
Comment (jeremiedimino) [inline Makefile:133]: There is already a `release` target in this Makefile

**Review by jeremiedimino — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (rgrinberg) [reply Makefile]: Ah damn. I reverted the name to `opam-release`.

**Review by jeremiedimino — APPROVED** [review]
Looks good

**Comment by rgrinberg** [comment]
@jeremiedimino some docs have been added to HACKING

**Comment by jeremiedimino** [comment]
Thanks, look good. BTW, IIRC semantic versioning correctly, incrementing `y` is called a "minor" or "feature" release. only incrementing `x` is called a major release. We should probably follow the official terminology to avoid confusion.

**Comment by rgrinberg** [comment]
Indeed. I pushed a fix.

**Pull request merged**